shared/src/androidMain/kotlin/com/karumi/gallery/domain/TimeProvider.kt

```diff
@@ -0,0 +1,29 @@
+package com.karumi.gallery.domain
+
+import android.content.SharedPreferences
+import java.util.Calendar
+
+actual class TimeProvider(
```

Comment:
I find it really odd that a class called `TimeProvider` stores anything. Provider usually means that its gets you something. I'd separate this class into two, `TimeProvider` having a `getCurrentTime` and `timeAgo` methods and a `TimeStorage` that can persist times.